key/key_test.go

```diff
@@ -49,6 +53,18 @@ func TestKeyEqual(t *testing.T) {
 		a:      New(map[string]interface{}{"a": 3}),
 		b:      New(map[string]interface{}{"a": 3}),
 		result: true,
+	}, {
+		a:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 3}}),
+		b:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 4}}),
+		result: false,
+	}, {
+		a:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 3, New("c"): 5}}),
+		b:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 4}}),
+		result: false,
+	}, {
+		a:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 4}}),
+		b:      New(map[string]interface{}{"a": map[Key]interface{}{New("b"): 4}}),
+		result: true,
```

Comment:
Add a two element equality case to confirm the iteration works for the positive case, but the code looks fine.